File: 125-valid-palindrome/valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        int n = s.length();
        int left = 0, right = n - 1;
        s = s.toLowerCase();
        while(left<right){
            char c1 = s.charAt(left);
            char c2 = s.charAt(right);

            if(!Character.isLetterOrDigit(c1)){
                left++;
                continue;
            }
            if(!Character.isLetterOrDigit(c2)){
                right--;
                continue;
            }

            if(c1 != c2){
                return false;
            }

            left++;
            right--;
        }

        return true;
    }
}

File: 125-valid-palindrome/valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s = ''.join(c for c in s if c.isalnum()).lower()
        
        n: int = len(s)
        right: int = n - 1

        for left in range(0, n):
            if(s[left] != s[right]): return False
            right -= 1
        
        return True